packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -164,34 +179,156 @@ class SkiaGoldClient extends GoldensClient {
+      final Uri requestForImage = Uri.parse(
+        'https://flutter-gold.skia.org/img/images/$imageHash.png',
+      );
+
+      try {
+        final io.HttpClientRequest request = await httpClient.getUrl(requestForImage);
+        final io.HttpClientResponse response = await request.close();
+        await response.forEach((List<int> bytes) => imageBytes.addAll(bytes));
+
+      } catch(e) {
+        rethrow;
+      }
+    },
+      SkiaGoldHttpOverrides(),
+    );
+    return imageBytes;
+  }
+
+  /// Returns a boolean value for whether or not the given test and current pull
+  /// request are ignored on Flutter Gold.
+  ///
+  /// This is only relevant when used by the [FlutterPreSubmitFileComparator].
+  /// In order to land a change to an exiting golden file, an ignore must be set
+  /// up in Flutter Gold. This will serve as a flag to permit the change to
+  /// land, and protect against any unwanted changes.
+  Future<bool> testIsIgnoredForPullRequest(String pullRequest, String testName) async {
+    bool ignoreIsActive = false;
+    testName = cleanTestName(testName);
+    String rawResponse;
+    await io.HttpOverrides.runWithHttpOverrides<Future<void>>(() async {
+      final Uri requestForIgnores = Uri.parse(
+        'https://flutter-gold.skia.org/json/ignores'
+      );
+
+      try {
+        final io.HttpClientRequest request = await httpClient.getUrl(requestForIgnores);
+        final io.HttpClientResponse response = await request.close();
+        rawResponse = await utf8.decodeStream(response);
+        final List<dynamic> ignores = json.decode(rawResponse);
+        for(Map<String, dynamic> ignore in ignores) {
+          final List<String> ignoredQueries = ignore['query'].split('&');
+          final String ignoredPullRequest = ignore['note'].split('/').last;
+          if (ignoredQueries.contains('name=$testName') &&
+            ignoredPullRequest == pullRequest) {
+            ignoreIsActive = true;
+            break;
+          }
+        }
+      } on FormatException catch(_) {
+        print('Formatting error detected requesting ignores from Flutter Gold.\n'
+          'rawResponse: $rawResponse');
+        rethrow;
+      }
+    },
+      SkiaGoldHttpOverrides(),
+    );
+    return ignoreIsActive;
+  }
+
+  /// The [_expectations] retrieved from Flutter Gold do not include the
+  /// parameters of the given test. This function queries the Flutter Gold
+  /// details api to determine if the given expectation for a test matches the
+  /// configuration of the executing machine.
+  Future<bool> isValidDigestForExpectation(String expectation, String testName) async {
+    bool isValid = false;
+    testName = cleanTestName(testName);
+    String rawResponse;
+    await io.HttpOverrides.runWithHttpOverrides<Future<void>>(() async {
+      final Uri requestForDigest = Uri.parse(
+        'https://flutter-gold.skia.org/json/details?test=$testName&digest=$expectation'
```

Comment:
FYI @kjlubick This appears to be the request that was timing out.

Reply:
Alright, that also is prone to the 5 QPS limit atm.  I'll dig in a bit and see what we can do about it before our discussion tomorrow.

Reply:
Actually, that one is on a rate limit that is way under 5QPS, but measurements show that it might not need to be.